### ISSUE TITLE
Can't resize, transform or augment training data with ObjectDetectionData.from_coco for object detection.

### ISSUE THREAD

**Issue by hzitoun** [PyTorchLightning/lightning-flash#409]
## 🐛 Bug

Getting errors when transforming or augmenting training data with ObjectDetectionData.from_coco for custom object detection.

### To Reproduce

I've a basic datamodule. I want to apply some random data augmentation using **albumentations** or **torchvision.transforms as T** but getting errors when fitting the model. For example, I can't even resize my images before training. 

```python
# using torchvision.transforms

image_size = 1024

datamodule = ObjectDetectionData.from_coco(
    train_folder="data_coco/train",
    train_ann_file="data_coco/train/_annotations.coco.json",
    train_transform={
      'pre_tensor_transform': T.Resize(image_size),
       'to_tensor_transform': T.ToTensor()
    },
    val_split=0.2,
    batch_size=32,
    num_workers=4,
)

model = ObjectDetector(model="retinanet", backbone="resnet101", num_classes=datamodule.num_classes, fpn=True)
trainer = flash.Trainer(max_epochs=10, limit_train_batches=1, limit_val_batches=1, gpus=1, checkpoint_callback=True)
trainer.finetune(model, datamodule=datamodule)


# using albumentations

train_transform = A.Compose(
    [
        A.RandomResizedCrop(height=image_size, width=image_size, scale=(0.9, 1), p=1),
        A.HorizontalFlip(p=0.5),
        A.ShiftScaleRotate(p=0.5),
        ToTensorV2()
    ],
     p=1.0, 
    bbox_params=A.BboxParams(
        format='coco',
        min_area=0, 
        min_visibility=0,
        label_fields=['labels']
    )
)

datamodule = ObjectDetectionData.from_coco(
    train_folder="data_coco/train",
    train_ann_file="data_coco/train/_annotations.coco.json",
    train_transform= train_transform,  #even tried to create a custom transform(sample) : -> transformed_sample but fails 
    val_split=0.2,
    batch_size=32,
    num_workers=4,
)
```

### Environment

 - PyTorch Version: 1.8
 - OS: Linux
 - How you installed PyTorch: !pip install -U lightning-flash !pip install -U lightning-flash[image] !pip install  -U albumentations
 - Python version: 3.7.10
 - CUDA/cuDNN version: 11
 - GPU models and configuration: Tesla P100 16GB

**Comment by ethanwharris** [comment]
Hi @hzitoun Thanks for the issue! Can you share the error you're getting? I suspect the issue is that the data for the `ObjectDetector` is a dictionary with `"input"` mapping to the image and `"target"` mapping to a bounding box dict (with `"boxes"`, `"labels"`, etc.). In order for resizing to work, you would have to create a function which maps those keys to the correct arguments for the transform. We do have `flash.core.data.transforms.ApplyToKeys` for simple cases where you just want to apply a transform to a particular key (or multiple keys). It's also worth noting that the torchvision object detection networks that we use will resize the images and boxes in the forward pass, so you shouldn't need to apply any transforms for your code to work.

I think there are a few things we can do to make this better:
- We should properly document transforms / augmentations including any specific details needed for each task.
- The object detector shouldn't rely on the model to convert the data into the correct format, we should do that in our pipeline.
- We should have better error messages, for example when the transforms provided are in the wrong format for the data.

@aniketmaurya any thoughts on this?

**Comment by aniketmaurya** [comment]
Yes @ethanwharris , I agree with the points you mentioned above for documentation.

May be in the [Tasks documentation](https://lightning-flash.readthedocs.io/en/latest/reference/object_detection.html) we can have simple example like the way it currently is and have a separate section **Examples** where we can have working examples with custom dataset, transforms.

**Comment by edgarriba** [comment]
@ethanwharris @aniketmaurya we recently included in kornia a `AugmentationSequential` that automatically to perform the augmentations in bounding boxes, masks, etc. We could try to integrate both things.

**Comment by ethanwharris** [comment]
Closing this as duplicated in #414

**Comment by hzitoun** [comment]
There is still no documentation or example showing how to resize the data before feeding them to the model! 
In case of high-resolution image, CUDA runs into OOM even with a tiny batch size. Resizing the image has to be done at data module level. I can't starting training a retinaNet with a V100 16GB! I need to resize the images.

**Comment by ethanwharris** [comment]
@hzitoun Yes but this is the same issue as #414 no? We should track this in one place to avoid confusion
